nnUnet/convert_msd_to_nnunet_reorient.py

```diff
@@ -0,0 +1,183 @@
+        # Process items in parallel and show progress bar
+        results = list(tqdm.tqdm(
+            pool.imap(process_single_image, work_items),
+            total=len(work_items),
+            desc="Processing images"
+        ))
+    
+    return results
+
+
+def main():
+    # Parse arguments
+    args = parse_args()
+    if args.workers is None:
+        args.workers = cpu_count()
+    
+    # Define the output paths
+    path_out = Path(os.path.join(args.output, f'Dataset{args.tasknumber}_{args.taskname}'))
+    path_out_imagesTr = Path(os.path.join(path_out, 'imagesTr'))
+    path_out_imagesTs = Path(os.path.join(path_out, 'imagesTs'))
+    path_out_labelsTr = Path(os.path.join(path_out, 'labelsTr'))
+    path_out_labelsTs = Path(os.path.join(path_out, 'labelsTs'))
+
+    # Create directories
+    for path in [path_out, path_out_imagesTr, path_out_imagesTs, path_out_labelsTr, path_out_labelsTs]:
+        path.mkdir(parents=True, exist_ok=True)
+
+    # Load datasets
+    datalists_list = [f for f in os.listdir(args.input) if f.endswith("_seed50.json")]
+    train_data, val_data, test_data = [], [], []
+    for datalist in sorted(datalists_list):
+        print(f"Loading dataset: {datalist}")
+        train_data += load_decathlon_datalist(os.path.join(args.input, datalist), True, "train")
+        val_data += load_decathlon_datalist(os.path.join(args.input, datalist), True, "validation")
+        test_data += load_decathlon_datalist(os.path.join(args.input, datalist), True, "test")
+
+    print(f"Processing {len(datalists_list)} datasets with {args.workers} workers...")
+    print(f"Number of training samples: {len(train_data)}")
+    print(f"Number of validation samples: {len(val_data)}")
+    print(f"Number of testing samples: {len(test_data)}")
+
+    # Process training data (including validation)
+    print("Processing training data...")
+    train_results = process_dataset_parallel(
+        train_data + val_data,
+        path_out_imagesTr,
+        path_out_labelsTr,
+        args.taskname,
+        1,
+        args.workers
+    )
+
+    # Process test data
+    print("Processing test data...")
+    test_results = process_dataset_parallel(
+        test_data,
+        path_out_imagesTs,
+        path_out_labelsTs,
+        args.taskname,
+        1,
+        args.workers
+    )
+
+    # Create conversion dictionary
+    conversion_dict = {}
+    for result in train_results + test_results:
+        conversion_dict[result['image']] = result['image_nnunet']
+        conversion_dict[result['label']] = result['label_nnunet']
+
+    # Save conversion dictionary
+    with open(os.path.join(path_out, "conversion_dict.json"), "w") as f:
+        json.dump(conversion_dict, f, indent=4)
+
+    # Create dataset description
+    json_dict = OrderedDict({
+        'name': args.taskname,
+        'description': args.taskname,
+        'tensorImageSize': "3D",
+        'reference': "TBD",
+        'licence': "TBD",
+        'release': "0.0",
```

Comment:
We could/should include `image_orientation` into the nnunet config file. To be read by SCT. As done, e.g., for SCIsegV2: https://github.com/ivadomed/model_seg_sci/blob/289473b7df1eba5d20a6ce8c5e2921091f9a12cc/dataset-conversion/convert_bids_to_nnUNetv2_all_sci_data.py#L221

Reply:
great catch! added in https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/pull/136/commits/cb51d558311772126894dfe60e9837dab4e45ccb